lib/langchain/processors/eml.rb

```diff
@@ -0,0 +1,59 @@
+require 'mail'
+require 'uri'
+
+module Langchain
+  module Processors
+    class Eml < Base
+      EXTENSIONS    = ['.eml']
+      CONTENT_TYPES = ['message/rfc822']
+
+      def initialize(*)
+        depends_on 'mail'
+      end
+
+      # Parse the document and return the cleaned text
+      # @param [File] data
+      # @return [String]
+      def parse(data)
+        mail         = Mail.read(data.path)
+        text_content = extract_text_content(mail)
```

Comment:
I've used this gem before, have you seen it? https://github.com/github/email_reply_parser